plugins/main/macvlan/macvlan.go

```diff
@@ -105,19 +107,45 @@ func createMacvlan(conf *NetConf, ifName string, netns ns.NetNS, configureNetwor
 	var mv netlink.Link
 
 	if configureNetworkForVM {
-		mv = &netlink.Macvtap{
-			Macvlan: netlink.Macvlan{
-				LinkAttrs: netlink.LinkAttrs{
-					MTU:         conf.MTU,
-					Name:        tmpName,
-					ParentIndex: m.Attrs().Index,
-					Namespace:   netlink.NsFd(int(netns.Fd())),
+		const hostLinkOffset = 8192
+		const linkRetries = 128
+		const linkRange = 0xFFFF
+
+		// There is a limitation in the linux kernel that prevents a macvtap/macvlan link
+		// from getting the correct link index when created in a network namespace
+		//
+		// Till that bug is fixed we need to pick a random non conflicting index and try to
+		// create a link. If that fails, we need to try with another.
+		// All the kernel does not check if the link id conflicts with a link id on the host
+		// hence we need to offset the link id to prevent any overlaps with the host index
+		//
+		// Here the kernel will ensure that there is no race condition
+
+		r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
+		for i := 0; i < linkRetries; i++ {
+			index := hostLinkOffset + (r.Int() & linkRange)
+			mv = &netlink.Macvtap{
+				Macvlan: netlink.Macvlan{
+					LinkAttrs: netlink.LinkAttrs{
```

Comment:
Why do we need to specify an interface index on creation?  We know the ifname so we can look up the index later if we need to, no?